azext_iot/deviceupdate/commands_update.py

```diff
@@ -0,0 +1,237 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from azext_iot.common.utility import handle_service_exception
+from azext_iot.deviceupdate.providers.base import (
+    DeviceUpdateDataModels,
+    DeviceUpdateDataManager,
+    AzureError,
+    ARMPolling,
+)
+from azure.cli.core.azclierror import ArgumentUsageError
+from typing import Optional, List, Union
+
+logger = get_logger(__name__)
+
+
+def list_updates(
+    cmd,
+    name,
+    instance_name,
+    search: Optional[str] = None,
+    filter: Optional[str] = None,
+    by_provider: Optional[bool] = None,
+    by_name: Optional[bool] = None,
+    by_version: Optional[bool] = None,
+    update_name: Optional[str] = None,
+    update_provider: Optional[str] = None,
+    resource_group_name: Optional[str] = None,
+):
+    data_manager = DeviceUpdateDataManager(
+        cmd=cmd, account_name=name, instance_name=instance_name, resource_group=resource_group_name
+    )
+    if [by_provider, by_name, by_version].count(True) > 1:
+        raise ArgumentUsageError("Only a single --by-* flag can be used at a time.")
+
+    try:
```

Comment:
How about a logger message that indicates the value is ignored if that condition is met? I see it as a simpler experience for the user if the client does not error because a user could provide update_name and update_provider then vary the pivot flags (--by-provider, --by-name, --by-version) without having to change arguments.

Reply:
Yeah logger message works well.